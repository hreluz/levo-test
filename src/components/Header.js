import PropTypes from 'prop-types';

const Header = ({title, paragraph}) => {
    return (
        <div className="row mt-2">
            <div className="col">
                <h1 className="text-danger">{ title }</h1>
                <p>{ paragraph }</p>
            </div>
        </div>
    );
}

Header.propTypes = {
    title: PropTypes.string.isRequired,
    paragraph: PropTypes.string.isRequired
}

export default Header;
