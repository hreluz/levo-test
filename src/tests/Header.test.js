import { shallow } from 'enzyme'
import Header from '../components/Header'

describe('Test in Header', () => {

    const title = 'A title';
    const paragraph = 'This is a paragraph to show for the test';

    const wrapper = shallow(<Header
                                paragraph={paragraph}
                                title={title}/>)

    test('it should show <Header/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('it should have a title in h1', () => {
        const h1 = wrapper.find('h1');
        expect(h1.text().trim()).toBe( title );
    });

    test('it should have a paragraph in p', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( paragraph );
    });
});