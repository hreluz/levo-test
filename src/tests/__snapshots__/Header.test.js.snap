// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Header it should show <Header/> correctly 1`] = `
<div
  className="row mt-2"
>
  <div
    className="col"
  >
    <h1
      className="text-danger"
    >
      A title
    </h1>
    <p>
      This is a paragraph to show for the test
    </p>
  </div>
</div>
`;
